matrix_messages.rst

```diff
@@ -0,0 +1,92 @@
+Raiden transport messages
+#########################
+
+Definitions
+===========
+* Sender: entity sending the message
+* Recipient: entity able of receiving and decoding the message
+* Message: A data object exchanged between Sender and Recipient
+* Message Field: an attribute of JSON data format that is required to be included in the message
+
+Transport envelope
+==================
+Messages are exchanged in JSON format.
+Message itself contains metadata that allow recipient to verify sender of the message (via use of PK signatures).
+
+Rationale
+---------
+* plaintext JSON messages are easy to read for both humans and machines
+* JSON format is widely used and can be easily validated using JSON-schema (http://json-schema.org/)
+* JSON is easily extensible and extension can be made backward-compatible
+* Message format must not be dependent on the transport used - it should not matter whether the message is sent via Matrix or REST
+* Sender identity verification must not be part of the transport. It should be possible to submit the message over an untrusted channel or delegate.
+
+Example of a message
+-------------------
+``
+{
+    'signature': '0xabcd....01234',
```

Comment:
Does a signature allow for check-summing of the message? Do you think that a checksum would be useful to have?

Reply:
we don't need that, the signature is also a checksum (data to sign are hashed first, then signed; vice versa when verifying the signature)

Reply:
Tthis is invalid json btw, json uses double quotes and not single quotes